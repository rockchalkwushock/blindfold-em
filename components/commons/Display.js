import SubTitle from './SubTitle'
import FlexContainer from './FlexContainer'
import Fragment from './Fragment'

const CoolDownDisplay = ({ base, current }) => (
  <Fragment>
    <SubTitle text={`Your selected cooldown: ${base}`} />
    <SubTitle text={`Current Cooldown: ${current}`} />
  </Fragment>
)

const TimerDisplay = ({ base, current }) => (
  <Fragment>
    <SubTitle text={`Your selected time: ${base}`} />
    <SubTitle text={`Current Time: ${current}`} />
  </Fragment>
)

const Display = ({
  activity,
  baseCool,
  baseTime,
  condition,
  currCool,
  currTime
}) => (
  <FlexContainer>
    <SubTitle text={`Your activity is: ${activity}`} />
    {!condition ? (
      <TimerDisplay base={baseTime} current={currTime} />
    ) : (
      <CoolDownDisplay base={baseCool} current={currCool} />
    )}
  </FlexContainer>
)

export default Display
