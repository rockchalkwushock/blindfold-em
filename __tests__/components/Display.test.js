import React from 'react'
import { render } from 'enzyme'

import Display from '../../components/commons/Display'

const props = {
  activity: 'coding',
  baseCool: '0:03:00',
  baseTime: '0:12:00',
  condition: null,
  currCool: '0:03:00',
  currTime: '0:12:00'
}

describe('Component: <Display />', () => {
  test('should render without exploding as <TimerDisplay /> ', () => {
    const tree = render(<Display {...props} />)
    expect(tree).toMatchSnapshot()
  })
  test('should render without exploding as <CoolDownDisplay /> ', () => {
    const tree = render(<Display {...props} condition={2} />)
    expect(tree).toMatchSnapshot()
  })
})
