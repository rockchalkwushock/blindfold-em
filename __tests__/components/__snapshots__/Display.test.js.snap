// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Display /> should render without exploding as <CoolDownDisplay />  1`] = `
<div>
  <h2>
    Your activity is: coding
  </h2>
  <h2>
    Your selected cooldown: 0:03:00
  </h2>
  <h2>
    Current Cooldown: 0:03:00
  </h2>
</div>
`;

exports[`Component: <Display /> should render without exploding as <TimerDisplay />  1`] = `
<div>
  <h2>
    Your activity is: coding
  </h2>
  <h2>
    Your selected time: 0:12:00
  </h2>
  <h2>
    Current Time: 0:12:00
  </h2>
</div>
`;
